extensions/cms/src/cmsUtils.ts

```diff
@@ -155,9 +157,12 @@ export class CmsUtils {
 				} else {
 					let registeredServerName = connection.options.registeredServerName === '' ? connection.options.server : connection.options.registeredServerName;
 					let result = await provider.addRegisteredServer(ownerUri, relativePath, registeredServerName, connection.options.registeredServerDescription, connection);
-					return result;
+					if (result) {
+						return Promise.resolve(result);
+					} else {
+						return Promise.reject(registeredServerName);
+					}
```

Comment:
Why not just return the awaited calls? Do you ever expect this to actually return an undefined result that isn't an error? I would generally say we should avoid doing that as it's not intuitive from an API standpoint

```
return await provider.createCmsServer(...);
```

Reply:
Yes, the backend sends undefined if it fails. The backend would need changes to send errors.